### PR TITLE
Added a Utility Module

### PR DESCRIPTION
**Pull request by Hibyehello**
Added functions:
    - Get the Script Directory (`get_script_dir()`)
    - Prompt user to open file using file dialog (`open_file()`)
    - Start and stop framedumps (`start_framedump()`, `stop_framedump()`)
    - Start and stop audiodumps (`start_audiodump()`, `stop_audiodump()`)
    - Check if frame/audio dumping is active (`is_framedumping()`, `is_audiodumping()`)
    - Save Screenshot (`save_screenshot(filename = Null)`)

We'll need to check compiling on windows ~, and create a stub.~ (done)

### CODE REVIEW

**Comment by vabold** [comment]
"Change text font (WIP)" is still in the commit history.

**Comment by Hibyehello** [comment]
Yes git decided to keep that, but I reverted it, so it's not actually there, we'll want to squash

**Comment by ximk** [comment]
This PR currently does not compile on Windows for me. I'd like someone else to try compiling on Windows, though.

File: Source/Core/Scripting/Python/Modules/utilmodule.h [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#pragma once
+
+#include <Python.h>
+
+namespace PyScripting
+{
+    PyMODINIT_FUNC PyInit_dol_utils();
+}
```
Comment (vabold) [inline Source/Core/Scripting/Python/Modules/utilmodule.h:8]: Newline.

**Review by vabold — CHANGES_REQUESTED** [review]

**Comment by Hibyehello** [comment]
I hope this works on windows, editing sln and vcxproj files by hand is confusing

File: Source/Core/Scripting/Python/Modules/utilmodule.cpp [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  std::string filePath;
+
+  NFD_Init();
+  
+  nfdchar_t *outPath;
+  nfdfilteritem_t filterItem[1] = { { "All Files", "*" } };
+  nfdresult_t result = NFD_OpenDialog(&outPath, filterItem, 1, NULL);
+
+  if(result != NFD_OKAY)
+  	PY_RETURN_NONE;
```
Comment (ximk) [inline Source/Core/Scripting/Python/Modules/utilmodule.cpp:91]: Fix capitalization.

**Comment by ximk** [comment]
Compiles on Windows if `nfd.vcxproj` is built independently of the main solution file.

File: Source/Core/Scripting/Python/Modules/utilmodule.h (hunk 1 above)
Comment (Hibyehello) [reply Source/Core/Scripting/Python/Modules/utilmodule.h]: I believe I resolved this, but I'm not sure exactly what you meant lol

File: Source/Core/Scripting/Python/Modules/utilmodule.cpp [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+static PyObject* save_screenshot(PyObject* module, PyObject* args, PyObject* kwargs)
+{
+  char* filename = nullptr;
+  static char* kwlist[] = {"filename", NULL};
+
+  if(!PyArg_ParseTupleAndKeywords(args, kwargs,"|s", kwlist, &filename))
+  	return nullptr;
+
+  if(filename != nullptr)
```
Comment (vabold) [inline Source/Core/Scripting/Python/Modules/utilmodule.cpp:73]:
```suggestion
  if(filename)
```

**Review by vabold — CHANGES_REQUESTED** [review]

**Review by vabold — APPROVED** [review]

**Pull request merged**